### PR TITLE
Enhancement enable more pyra tests

### PR DESCRIPTION
**Pull request by Davknapp**
Three tests weren't enabled for pyramids. This PR enables them for pyramids.

**_All these boxes must be checked by the reviewers before merging the pull request:_**

- [x] The author added a BSD statement to `doc/` (or already has one)
- [x] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)

  If the Pull request indroduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

- [x] All tests pass (in various configurations, this should be executed automatically in a github action)
- [ ] New source/header files are properly added to the Makefiles
- [x] The reviewer executed the new code features at least once and checked the results manually
- [x] The code is covered in an existing or new test case
- [x] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [x] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)
- [ ] Testing of this template: If you feel something is missing from this list, contact the developers

### CODE REVIEW

File: src/t8_schemes/t8_default/t8_default_pyramid/t8_default_pyramid_cxx.hxx [hunk 1]
```diff
@@ -308,6 +308,18 @@ public:
                                                           *elem, int vertex,
                                                           double coords[]);
 
+  /** The implementation of the general function for the default scheme for pyramids.
+   * Stores the type of the pyramid in \a outdata.
+   *  \param [in] elem A valid element
+   *  \param [in] indata Is ignored. Can be NULL.
+   *  \param [out] outdata The type of \a elem.
+   * \note Calling this function has no effect. See the specialized implementations in
+   * t8_default_tri_cxx.hxx, t8_default_tet_cxx.hxx and t8_default_prism_cxx.hxx.
```
Comment (lukasdreyer) [inline src/t8_schemes/t8_default/t8_default_pyramid/t8_default_pyramid_cxx.hxx:317]: Copy paste leftover from the empty default implementation?

File: test/t8_geometry/t8_test_point_inside.cxx [hunk 2]
```diff
@@ -262,7 +262,7 @@ main (int argc, char **argv)
     ("Testing one specific point with one specific triangle.\n");
   t8_test_point_inside_specific_triangle ();
   for (ieclass = T8_ECLASS_LINE; ieclass < T8_ECLASS_COUNT; ieclass++) {
-    if (ieclass != T8_ECLASS_PYRAMID && ieclass != T8_ECLASS_QUAD) {
+    if (ieclass != T8_ECLASS_QUAD) {
       /* TODO: - does not work with pyramids yet, since pyramid elements are not implemented.
        *         the point check should work with pyramids, once they are implemented.
        *       - point inside check does not work with quads yet. */
```
Comment (lukasdreyer) [inline test/t8_geometry/t8_test_point_inside.cxx:268]:
```suggestion
      /* TODO: point inside check does not work with quads yet. */
```
Do you know why the point check does not yet work with quads?

**Review by lukasdreyer — CHANGES_REQUESTED** [review]

File: test/t8_geometry/t8_test_point_inside.cxx (hunk 2 above)
Comment (Davknapp) [reply test/t8_geometry/t8_test_point_inside.cxx:268]: The function t8_forest_element_point_inside is not implemented for quds

Comment (lukasdreyer) [reply test/t8_geometry/t8_test_point_inside.cxx]: Why is that more complicated than for tris and tets? Do we plan to implement such a function, i.e. is there an Issue for it?

**Review by lukasdreyer — APPROVED** [review]
So nice to see pyramids working now!

**Pull request merged**